### DISCUSSION TITLE
Payload builder new payload compile error

### DISCUSSION THREAD

**Discussion by scottmillner** [paradigmxyz/reth#8232] [Help] [answered]
I'm requesting a new payload to be built, I'm bypassing the Engine API and directly calling the `payload_builder.new_payload()` method like so:

```
let rpc_payload_attributes = PayloadAttributes {
            timestamp: 10000,
            prev_randao: B256::ZERO,
            suggested_fee_recipient: Address::default(),
            withdrawals: None,
            parent_beacon_block_root: None,
        };
let attr = PayloadBuilderAttributes::try_new(B256::ZERO, rpc_payload_attributes).unwrap();
let payload_builder = components.payload_builder().clone();
payload_builder.new_payload(attr);
```

I get the following error:

```
note: expected associated type `<<T as reth_node_api::NodeTypes>::Engine as reth_node_api::EngineTypes>::PayloadAttributes`
found struct reth_rpc_types::alloy_rpc_types_engine::PayloadAttributes
```

The error can be reproduced in `crates/node/builder/src/launch/mod.rs` by placing the code anywhere after:
`let components = components_builder.build_components(&builder_ctx).await?;`
                        
I tried wiring up `Cli::parse_args().run`
in `bin/reth/src/main` based on the example in `examples/custom-payload-builder` but to no success.

I would greatly appreciate any help in this matter.

**Comment by mattsse** [comment, marked as answer]
are you modifying the existing launch function?

using specific types won't work here because this is generic over enginetypes and not ethereum engine types

ptal here:

https://github.com/paradigmxyz/reth/compare/matt/add-payload-builder-example?expand=1

**Reply by scottmillner** [reply]
yes, trying to modify launch function but can't be done

i'll use a different approach

thanks for the quick response!
